src/libsync/syncfilestatustracker.cpp

```diff
@@ -177,6 +181,7 @@ void SyncFileStatusTracker::slotAboutToPropagate(SyncFileItemVector& items)
 
     foreach (const SyncFileItemPtr &item, items) {
         // qDebug() << Q_FUNC_INFO << "Investigating" << item->destination() << item->_status << item->_instruction;
+        _dirtyPaths.remove(item->destination());
```

Comment:
It's related to the `oldDirtyPaths` logic lower, we need to know what file detected by the filesystemwatcher will actually be propagated (vs. e.g. checksum checks by csync).